### DISCUSSION TITLE
PLease provide any feedback , suggestions or start a chat about features you would like to see or ways we can improve this project.

### DISCUSSION THREAD

**Discussion by BigBobbas** [BigBobbas/ESP32-S3-Box3-Custom-ESPHome#6] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by rcblackwell** [comment]
@BigBobbas, the [Installation Guide Note](https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/blob/main/instructions/installation%20guide.md#note) indicates users may experience issues when attempting to compile the code on a machine with low resources. I'm running an HA Blue with 8G that would never compile code for the S3-Box until I limited the number of processors used during compile. I've set the limit to 1. Sure, compiling the code takes longer but at least the machine doesn't crash. You may wish to mention that in the **Notes:** section. Here's what I've added;

```
esphome:
  compile_process_limit: 1
```

**Reply by BigBobbas** [reply]
Thank you for the suggestion. I will add this as an option to try in the notes. 
Always good to know what has worked for others. Thank you.

**Reply by BigBobbas** [reply]
added .... https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/edit/main/instructions/installation%20guide.md#note

**Comment by rhyswaywood** [comment]
I have an ESP32-S3-Box-3B without the sensor dock. What changes would I need to make to the config for this to run?

**Reply by BigBobbas** [reply]
Thats a good question, I completely forgot about the non-sensor version. I will have a think and a tweak and see if i can come up with a config,  as it would also be useful to others too. I'm not too sure if it will work without the battery as this causes an issue with the touch and voice not working in harmony and everything goes belly up... However it is worth me testing. 
i'll drop a note in here once i've got something together

**Reply by BigBobbas** [reply]
I have put this together if you wanted to give it a test, it is working for me. [S3 Box - No Sensors](<https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/blob/main/s3b_no_sensors.yaml>)

**Reply by rhyswaywood** [reply]
It is working for me too! Thank you so much!!

**Reply by BigBobbas** [reply]
Great stuff! I will add a seperate link to it in the docs too, glad you mentioned it.

**Reply by rhyswaywood** [reply]
The device is so much better to use now it has volume control. The TTS responses were way too quiet before. Thank you so much for your work on this. I'm looking forward to customising it further in the next few days.

**Reply by IanSMoyes** [reply]
Your code is FABULOUS! As are your instructions/suggestions for "make it your own". Thank you so much for this.

**Reply by BigBobbas** [reply]
> Your code is FABULOUS! As are your instructions/suggestions for "make it your own". Thank you so much for this.

Thank you very much for your comments. It makes it all worthwhile - cheers

**Reply by IanSMoyes** [reply]
> > Your code is FABULOUS! As are your instructions/suggestions for "make it your own". Thank you so much for this.
> 
> Thank you very much for your comments. It makes it all worthwhile - cheers

You're welcome Rob. A little food for thought:
1: There's no mention of the ESP32-S3's IMU in your code. How do I "get to it"?
2: The "status bar" at the top of the screen is redefined at the start of each page definition. Is there a way to page the status bar definition into a "script" then call the script at the beginning of each page definition?
3: Can the icon definitions be formatted as a dictionary, like the colors (I'm a Brit too) are?

**Reply by BigBobbas** [reply]
You're welcome Rob. A little food for thought: 1: There's no mention of the ESP32-S3's IMU in your code. \\\\\\\\\\\\\\\\\
>I will be honest I didn't know that there was an IMU lol , I will look into that.

 2: The "status bar" at the top of the screen is redefined at the start of each page definition. Is there a way to page the status bar.

>This is something I can look into, I am not a coder so this is something i've not done before, other than using !include and having that part of the config into a separate file. I have left everything all in one config for ease of troubleshooting whilst still building and tweaking.

3: Can the icon definitions be formatted as a dictionary, like the colors (I'm a Brit too) are?
>again , this is something I can look into too as it's not something I am familiar with.

Thank you so much for your suggestions, it's great to get other eyes on it so that we can improve the experience for everyone, and make improvements in functionality and efficiently. 
Cheers - Rob

**Comment by youkorr** [comment]
hello I have 2 problems on my esp32BOX3 the first if I ask voice assistant a request after the three requests it remains blocked I have to reboot it
Then the screen saver gives incorrect information the time is not syncronized you have to press the home button for it to syncronize

Have the same problem as me

**Comment by youkorr** [comment]
I'm going to do a clean install to see if it's mine

**Comment by youkorr** [comment]
After cleaning the firmware the screen saver works better and everything is synchronized but for the voice part I have the same problem it remains blocked after a few requests I have to reboot it
![Capture d’écran 2024-06-12 180139](https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/assets/96953496/1840ed44-7dc3-45ee-8c86-636bacce8e5e)

**Reply by BigBobbas** [reply]
what version of ESPHome do you have installed? it will show this at the bottom of the dashboard. I have a fix in place for the time not updating. I will publish this at some point in the next day

**Comment by youkorr** [comment]
Current version: 2024.5.5

**Comment by youkorr** [comment]
ok I can't wait to install it :)

**Reply by BigBobbas** [reply]
that should be merged now, so you can use https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/blob/main/s3b.yaml
I have added an option to use the radar sensor to also wake the screen. See how yo get on 👍

**Comment by youkorr** [comment]
Ok I'll test it and get back to you

**Comment by youkorr** [comment]
too bad we can't use the SD card reader, reading would be quick

**Comment by youkorr** [comment]
I installed it but I couldn't find an option to wake up the screen

**Comment by youkorr** [comment]
sorry I didn't install the correct firmware

**Comment by youkorr** [comment]
I replaced this

it.printf(10, 175, id(my_font5), yellow, "wake on presecence :"); 
to 
it.printf(10, 175, id(my_font5), yellow, "wake on presence :"); 

I really like this firmware because of the voice bug which blocks you either stop media player or reboot

**Reply by BigBobbas** [reply]
i will update thanks

**Comment by styphonthal** [comment]
When I do multiple "okay nabu" commands in a row (around 3), it will stop replying to "okay nabu" unless I touch the right bottom icon and then main menu back. This occurs with the wakeup word being on device or on home assistant.

**Reply by BigBobbas** [reply]
Hi thanks for letting me know ,
is that with on device or home assistant wake word ?
cheers

**Reply by styphonthal** [reply]
both

**Reply by youkorr** [reply]
me too it remains blocked I haven't found where it comes from

**Reply by BigBobbas** [reply]
A fix to try here for Voice not responding after x commands.

https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/discussions/10

**Comment by youkorr** [comment]
good evening how to reduce the standby time

**Reply by BigBobbas** [reply]
![image](https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/assets/150487209/bc989bfb-6509-4b2b-992f-8b8d381a22de)
the 180s is the time to go to blank - if you have blank switched on , the 30s delay is the initial delays before the time is displayed.
The saver_page is where the displayed time and temp are set if you wanted to change anything there 
![image](https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/assets/150487209/6b87f53e-7722-418b-8423-45cd68650603)

**Comment by youkorr** [comment]
Thank you

**Comment by youkorr** [comment]
Did you find the problem that voice assistant blocks?

**Reply by BigBobbas** [reply]
can you explain in a little more detail, do you mean that the voice assistant stops responding? and if so , when does it happen? or how often? also if you can view the logs from the ESPHome dashboard and see if anything shows in the logs at the points before, during and after it fails. Thanks

**Comment by youkorr** [comment]
[logs_s3b-radar-sensor_logs.txt](https://github.com/user-attachments/files/15837097/logs_s3b-radar-sensor_logs.txt)

**Comment by youkorr** [comment]
it stops responding after the second or third it's random the standby no longer works, you have to reboot for everyone, it works again

**Reply by styphonthal** [reply]
I don't have to reboot, I just click on the command icon, then main menu back with red button. Doing this once or twice let's me use wake word again.

**Reply by BigBobbas** [reply]
working on a fix for this , sorry for the delay

**Reply by youkorr** [reply]
have you seen the logs that I sent to you

**Comment by youkorr** [comment]
Hello I would like to insert the time and temp into a screen saver on this firmawre made by Lajos similar to the bigbobbas project could you tell me how to do it thank you
https://github.com/youkorr/S3-Box-3-firmware-fork/blob/main/radar

You can test it

**Reply by BigBobbas** [reply]
sorry I don't have time to work on a different firmware, have you asked the author of that firmware ?

**Reply by youkorr** [reply]
I asked him I have been waiting for a while for his response to ask for his permission

**Reply by IanSMoyes** [reply]
> I asked him I have been waiting for a while for his response to ask for his permission

I feel you are being a little demanding. Rob might be too polite to say so but he does this for a hobby, in his free time, he also has a life too.

**Reply by youkorr** [reply]
yes it's true I'm sorry

**Reply by youkorr** [reply]
I respect his work a lot, I'm sorry for asking him for this service.

**Comment by mikey60** [comment]
**Thanks for a great program!**  I am slowly making modifications to personalize it. I am seriously considering using your program on all my ESP32-S3-Box-3 devices that now run Willow.  Especially since there is a Willow Inference Server (WIS) SST custom integration that I direct to my old Windows computer (now Linux server) that runs the WIS with a GPU.   The SST of WIS seems to be a little more accurate and faster than Faster Whisper SST that is running on the same Server.
